python/cudf/cudf/core/frame.py

```diff
@@ -3702,6 +3395,8 @@ def _merge(
         elif how in {"leftsemi", "leftanti"}:
             merge_cls = MergeSemi
 
+        # TODO: the two isinstance checks below indicates that `_merge` should
```

Comment:
is there a refactoring issue # around this? might be able to knock this out pretty quick.

Reply:
Not yet - I was thinking to completely get rid of the `isinstance` check, we might need to adopt double dispatch pattern. Something along the lines:
```python
class IndexedFrame:
    def _merge(rhs, **args, **kwargs):
        return rhs._rmerge_indexed_frame(self, lhs_is_index=False, *args, **kwargs)
    def _rmerge_indexed_frame(self, lhs: IndexedFrame, *args, **kwargs):
        return merge_cls(lhs, self, rhs_is_index=False, *args, **kwargs)
    def _rmerge_index(self, lhs: _base_index, **args, **kwargs):
        return merge_cls(lhs, self, rhs_is_index=False, *args, **kwargs)

class _base_index:
    def _merge(rhs, **args, **kwargs):
        return rhs._rmerge_index(self, rhs_is_index=True, *args, **kwargs)
    def _rmerge_indexed_frame(self, rhs: IndexedFrame, *args, **kwargs):
        return merge_cls(lhs, self, rhs_is_index=True, *args, **kwargs)
    def _rmerge_index(self, lhs: _base_index, **args, **kwargs):
        return merge_cls(lhs, self, rhs_is_index=True, *args, **kwargs)
```

Reply:
Maybe we can wrap up the duplicate logics in a `_merge_mixin` or something.

Reply:
Discussed offline, we're not going to try and make this change in this PR but will revisit it later.